sdk/samples/iot/vxworks/azureProvisioningClientSample.c

```diff
@@ -0,0 +1,996 @@
+  if (az_result_succeeded(azureRes))
+  {
+    printf_s("az_iot_provisioning_client_init OK!");
+  }
+  else
+  {
+    printf_s("az_iot_provisioning_client_init ERROR!\n");
+    return false;
+  }
+
+  azureRes = az_iot_provisioning_client_get_client_id(
+      azureClient, azureClientId, azureClientIdLen, NULL);
+  if (az_result_succeeded(azureRes))
+  {
+    printf_s("az_iot_provisioning_client_get_client_id OK: %s\n", azureClientId);
+  }
+  else
+  {
+    printf_s("az_iot_provisioning_client_get_client_id ERROR!\n");
+    return false;
+  }
+
+#ifndef X509_CERTIFICATE
+  /* generate a password for MQTT */
+  if (!mqttPasswordGen(azureClient, pwd, sizeof(pwd)))
+  {
+    return false;
+  }
+#endif
+
+  azureRes = az_iot_provisioning_client_get_user_name(
+      azureClient, azureUserName, azureUserNameLen, NULL);
+  if (az_result_succeeded(azureRes))
+  {
+    printf_s("az_iot_provisioning_client_get_user_name OK: %s\n", azureUserName);
+  }
+  else
+  {
+    printf_s("az_iot_provisioning_client_get_user_name ERROR!\n");
+    return false;
+  }
+
+  return true;
+}
+
+/*******************************************************************************
+ *
+ * mqttClientDestroy - destroy a MQTT client object
+ *
+ * This routine firstly disconnects the MQTT client from the Azure IoT Hub
+ * provisioning service, destroys the mosquitto object and do the library
+ * cleanup for mosquitto.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: N/A
+ *
+ * \NOMANUAL
+ */
+
+static void mqttClientDestroy(struct mosquitto* mqttClient /* MQTT client object */
+)
+{
+  if (mqttClient != NULL)
+  {
+    (void)mosquitto_disconnect(mqttClient);
+    mosquitto_destroy(mqttClient);
+    mqttClient = NULL;
+    (void)mosquitto_lib_cleanup();
+  }
+}
+
+/*******************************************************************************
+ *
+ * mqttClientCreate - create a MQTT client object
+ *
+ * This routine creates a MQTT client object and connects it to the
+ * Azure IoT Hub provisioning service.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: a pointer to the MQTT client object if successful, otherwise NULL
+ *
+ * \NOMANUAL
+ */
+
+static struct mosquitto* mqttClientCreate(
+    az_iot_provisioning_client* azureClient, /* provisioning client obj */
+    char* azureUserName, /* user name */
+    char* azureClientId /* client ID */
+)
+{
+  int mqttRes;
+  struct mosquitto* mqttClient = NULL;
+
+  /* init the mosquitto lib */
+  mqttRes = mosquitto_lib_init();
+  if (mqttRes == MOSQ_ERR_SUCCESS)
+  {
+    printf_s("mosquitto_lib_init() OK\n");
+  }
+  else
+  {
+    printf_s("mosquitto_lib_init ERROR: %s\n", mosquitto_strerror(mqttRes));
+    return NULL;
+  }
+
+  /* create a client */
+  mqttClient = mosquitto_new(azureClientId, true, &azureClient);
+  if (mqttClient != NULL)
+  {
+    printf_s("mosquitto_new OK!\n");
+  }
+  else
+  {
+    printf_s("mosquitto_new ERROR: %s\n", mosquitto_strerror(mqttRes));
+    (void)mosquitto_lib_cleanup();
```

Comment:
why? mqttClient will be NULL

Reply:
Do you mean why to call "mosquitto_lib_cleanup"?
Because "mosquitto_lib_cleanup" is corresponding to "mosquitto_lib_init".
It has no relationship with mqttClient.